### ISSUE TITLE
Collapsible if reported when the outer if body contains a cfg flag

### ISSUE THREAD

**Issue by janTatesa** [rust-lang/rust-clippy#16715]
### Summary

If the code is cfg'ed out it is perceived as if it wasn't there and therefore this error is reported

### Reproducer

```rust
            if let Some(ch) = &mut *child {
                if ch.try_wait()?.is_some() {
                    print_watching();
                    *child = None;
                }
                #[cfg(feature = "audio")]
                {
                    let mut buf = [0; 8];
                    if audio_time_socket.recv(&mut buf).is_ok_and(|n| n == 8) {
                        let time = f64::from_be_bytes(buf);
                        audio_time.store(time.to_bits(), Ordering::Relaxed);
                    }
                }
            }
```

### Version

```text
rustc 1.96.0-nightly (1d8897a4e 2026-03-13)
binary: rustc
commit-hash: 1d8897a4e88051480b36c501e66809425c6e08ad
commit-date: 2026-03-13
host: x86_64-unknown-linux-gnu
release: 1.96.0-nightly
LLVM version: 22.1.0
```
